cmd/osm-controller/osm-controller.go

```diff
@@ -245,22 +246,34 @@ func main() {
 		events.GenericEventRecorder().FatalEvent(err, events.InitializationError, "Error initializing ADS server")
 	}
 
-	// Initialize function and HTTP health probes
-	funcProbes := []health.Probes{xdsServer}
-	httpProbes := getHTTPHealthProbes()
-
-	// Initialize the http server and start it
-	httpServer := httpserver.NewHTTPServer(funcProbes, httpProbes, metricsstore.DefaultMetricsStore, constants.MetricsServerPort)
-	httpServer.Start()
-
 	if err := createControllerManagerForOSMResources(certManager); err != nil {
 		events.GenericEventRecorder().FatalEvent(err, events.InitializationError, "Error creating controller manager to reconcile OSM resources")
 	}
 
-	// Expose /debug endpoints and data only if the enableDebugServer flag is enabled
+	// Initialize OSM's http service server
+	httpServer := httpserver.NewHTTPServer(constants.OSMServicePort)
+
+	// Health/Liveness probes
+	funcProbes := []health.Probes{xdsServer}
+	httpServer.AddHandlers(map[string]http.Handler{
+		"/health/ready": health.ReadinessHandler(funcProbes, getHTTPHealthProbes()),
+		"/health/alive": health.LivenessHandler(funcProbes, getHTTPHealthProbes()),
+	})
+	// Metrics
+	httpServer.AddHandler("/metrics", metricsstore.DefaultMetricsStore.Handler())
+	// Version
+	httpServer.AddHandler("/version", version.GetVersionHandler())
+
+	// Start HTTP server
+	err = httpServer.Start()
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Failed to start OSM metrics/probes HTTP server")
+	}
+
+	// Create DebugServer and start its config event listener.
+	// Listener takes care to start and stop the debug server as appropriate
 	debugConfig := debugger.NewDebugConfig(certDebugger, xdsServer, meshCatalog, kubeConfig, kubeClient, cfg, kubernetesClient)
-	debugServerInterface := httpserver.NewDebugHTTPServer(debugConfig, constants.DebugPort)
-	httpserver.RegisterDebugServer(debugServerInterface, cfg)
+	debugConfig.StartDebugServerConfigListener()
```

Comment:
To make this consistent with HttpServer init, I would redo this code block as follows:
```
debugServer := httpserver.NewHTTPServer(constants.DebugPort)
debugger.Start(debugServer, debugConfig)
```

Reply:
only to make sure, this is not an httpserver. DebugConfig has the httpserver inside.
Starting or not the httpserver is up to DebugConfig, and should not really be done here, if I understood your suggestion properly.

Reply:
> `DebugConfig has the httpserver inside`

This logically does not make much sense, for a config to have a server inside? This is why I am suggesting to redo this part a bit differently.

Reply:
starting or not the server will depend on the config, you will need to have a handle to the server to issue the calls.
this call starts the `pkg/debugger/debugger_config_listener.go` function which monitors the configmap for changes on `enable_debug_server`

I'll listen to suggestions if you have other ideas, ofc

Reply:
I get that the config is required, but I think it would be more appropriate encapsulate the debugConfig within the debugServer instead of the other way around.